### PR TITLE
feat: API changes for integration + helm improvement (MAPCO-2931)

### PR DESCRIPTION
**Pull request by ronenkapelian**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✔                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                       |
| Chore            | ✔                                                                     |

Update some of values as part of preparation to client's integration

### CODE REVIEW

File: openapi3.yaml [hunk 1]
```diff
@@ -303,6 +306,22 @@ components:
...
+          type: string
+          format: uuid
+        taskIds:
+          type: array
+          items:
+            type: string
+            format: uuid
+      required:
+        - jobId
+        - taskIds
```
Comment (asafMasa) [inline openapi3.yaml:324]: shouldn't "naiveCacheJobResponse" contain the description?

File: src/createPackage/models/createPackageManager.ts [hunk 2]
```diff
@@ -610,8 +612,8 @@ export class CreatePackageManager {
 
   private async checkForExportDuplicate(
     dupParams: JobExportDuplicationParams,
-    callbackUrls: ICallbackTargetExport[]
-  ): Promise<ICallbackExportResponse | ICreateJobResponse | undefined> {
+    callbackUrls: ICallbackTargetExport[] | undefined
```
Comment (asafMasa) [inline src/createPackage/models/createPackageManager.ts]: callbackUrls: ICallbackTargetExport[] | undefined -> callbackUrls?: ICallbackTargetExport[]

File: src/createPackage/models/createPackageManager.ts [hunk 3]
```diff
@@ -675,16 +677,18 @@ export class CreatePackageManager {
 
   private async checkForExportProcessing(
     dupParams: JobExportDuplicationParams,
-    newCallbacks: ICallbackTargetExport[]
-  ): Promise<ICreateJobResponse | undefined> {
+    newCallbacks: ICallbackTargetExport[] | undefined
```
Comment (asafMasa) [inline src/createPackage/models/createPackageManager.ts]: newCallbacks: ICallbackTargetExport[] | undefined -> newCallbacks?: ICallbackTargetExport[]

File: src/createPackage/models/createPackageManager.ts [hunk 4]
```diff
@@ -675,16 +677,18 @@ export class CreatePackageManager {
...
+  ): Promise<ICreateExportJobResponse | undefined> {
     this.logger.info({ ...dupParams, roi: undefined, msg: `Checking for PROCESSING duplications with parameters` });
     const processingJob =
       (await this.jobManagerClient.findExportJob(OperationStatus.IN_PROGRESS, dupParams, true)) ??
       (await this.jobManagerClient.findExportJob(OperationStatus.PENDING, dupParams, true));
     if (processingJob) {
-      await this.updateExportCallbackURLs(processingJob, newCallbacks);
+      if (newCallbacks) {
+        await this.updateExportCallbackURLs(processingJob, newCallbacks);
```
Comment (asafMasa) [inline src/createPackage/models/createPackageManager.ts]: you can insert the condition inside the updateExportCallbackURLs() method, and if newCallbacks will be undefin return;

File: src/tasks/models/tasksManager.ts [hunk 5]
```diff
@@ -153,19 +153,21 @@ export class TasksManager {
...
-        const params: ICallbackExportData = { ...callbackParams, roi: job.parameters.roi };
-        callbackPromises.push(this.callbackClient.send(target.url, params));
-      }
-
-      const promisesResponse = await Promise.allSettled(callbackPromises);
-      promisesResponse.forEach((response, index) => {
-        if (response.status === 'rejected') {
-          // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-          this.logger.error({ reason: response.reason, url: targetCallbacks[index].url, jobId: job.id, msg: `Failed to send callback to url` });
+      if (targetCallbacks) {
```
Comment (asafMasa) [inline src/tasks/models/tasksManager.ts]: change the condition to:
```
if (!targetCallbacks) {
    return;
}
// else
```

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: src/createPackage/models/createPackageManager.ts (hunk 2 above)
Comment (ronenkapelian) [reply src/createPackage/models/createPackageManager.ts]: done

File: src/createPackage/models/createPackageManager.ts (hunk 3 above)
Comment (ronenkapelian) [reply src/createPackage/models/createPackageManager.ts]: done

File: src/tasks/models/tasksManager.ts (hunk 5 above)
Comment (ronenkapelian) [reply src/tasks/models/tasksManager.ts]: done

File: src/createPackage/models/createPackageManager.ts (hunk 4 above)
Comment (ronenkapelian) [reply src/createPackage/models/createPackageManager.ts]: done

File: src/tasks/models/tasksManager.ts (hunk 5 above)
Comment (asafMasa) [reply src/tasks/models/tasksManager.ts]: remove the "else {" as it's not needed

**Review by asafMasa — APPROVED** [review]

**Pull request merged**